core/src/main/java/com/yahoo/oak/OakMap.java

```diff
@@ -28,60 +28,58 @@
         implements AutoCloseable, ConcurrentNavigableMap<K, V>, ConcurrentZCMap<K, V> {
 
     private final InternalOakMap<K, V> internalOakMap;
-    /*
-     * Memory manager cares for allocation, de-allocation and reuse of the internally pre-allocated
-     * memory. There can be separate memory managing algorithms for keys and values.
-     * */
-    //private final MemoryManager valuesMemoryManager;
-    //private final MemoryManager keysMemoryManager;
+
+    // Used for iterators
     private final OakTransformer<K> keyDeserializeTransformer;
     private final OakTransformer<V> valueDeserializeTransformer;
     private final Function<Map.Entry<OakScopedReadBuffer, OakScopedReadBuffer>,
             Map.Entry<K, V>> entryDeserializeTransformer;
+
     private final OakComparator<K> comparator;
 
     // SubOakMap fields
     private final K fromKey;
     private final boolean fromInclusive;
     private final K toKey;
-    private boolean toInclusive;
+    private final boolean toInclusive;
     private final boolean isDescending;
 
-    // internal constructor, to create OakMap use OakMapBuilder
-    OakMap(K minKey, OakSerializer<K> keySerializer, OakSerializer<V> valueSerializer, OakComparator<K> oakComparator,
-        int chunkMaxItems, MemoryManager vMM, MemoryManager kMM) {
-
+    private OakMap(InternalOakMap<K, V> internalOakMap,
+                   K fromKey, boolean fromInclusive, K toKey,
+                   boolean toInclusive, boolean isDescending) {
+        OakSharedConfig<K, V> config = internalOakMap.config;
 
-        this.comparator = oakComparator;
-        this.internalOakMap = new InternalOakMap<>(minKey, keySerializer, valueSerializer, oakComparator,
-                vMM, kMM, chunkMaxItems, new ValueUtils());
+        this.internalOakMap = internalOakMap;
 
-        this.fromKey = null;
-        this.fromInclusive = false;
-        this.toKey = null;
-        this.isDescending = false;
+        this.fromKey = fromKey;
+        this.fromInclusive = fromInclusive;
+        this.toKey = toKey;
+        this.toInclusive = toInclusive;
+        this.isDescending = isDescending;
 
-        this.keyDeserializeTransformer = keySerializer::deserialize;
-        this.valueDeserializeTransformer = valueSerializer::deserialize;
+        this.comparator = config.comparator;
+        this.keyDeserializeTransformer = config.keySerializer::deserialize;
+        this.valueDeserializeTransformer = config.valueSerializer::deserialize;
         this.entryDeserializeTransformer = entry -> new AbstractMap.SimpleEntry<>(
-                keySerializer.deserialize(entry.getKey()),
-                valueSerializer.deserialize(entry.getValue()));
+                keyDeserializeTransformer.apply(entry.getKey()),
+                valueDeserializeTransformer.apply(entry.getValue()));
+    }
+
+    // internal constructor, to create OakMap use OakMapBuilder
+    OakMap(OakSharedConfig<K, V> config, K minKey, int chunkMaxItems) {
```

Comment:
Where this construct is used? Should it be private?

Reply:
Used by the builder. Added comment.